tests/gis_tests/distapp/tests.py

```diff
@@ -475,12 +473,16 @@ def test_distance_function_raw_result_d_lookup(self):
         with self.assertRaisesMessage(ValueError, msg):
             list(qs)
 
-    @no_oracle  # Oracle already handles geographic distance calculation.
     @skipUnlessDBFeature("has_Distance_function", 'has_Transform_function')
     def test_distance_transform(self):
         """
         Test the `Distance` function used with `Transform` on a geographic field.
         """
+        if connection.vendor == 'oracle':
+            # Disable FuncTestMixin check as OracleSpatialAdapter mutates
+            # polygons.
+            self.func_patcher.stop()
```

Comment:
We cannot call `stop()` more then once, it causes an error in `tearDown()`:
```
RuntimeError: stop called on unstarted patcher
```
We could add a small workaround: :thinking: 
```diff
diff --git a/tests/gis_tests/distapp/tests.py b/tests/gis_tests/distapp/tests.py
index 10227fe328..6b9890d3a8 100644
--- a/tests/gis_tests/distapp/tests.py
+++ b/tests/gis_tests/distapp/tests.py
@@ -482,6 +482,7 @@ class DistanceFunctionsTests(FuncTestMixin, TestCase):
             # Disable FuncTestMixin check as OracleSpatialAdapter mutates
             # polygons.
             self.func_patcher.stop()
+            self.func_patcher = None
 
         # We'll be using a Polygon (created by buffering the centroid
         # of 77005 to 100m) -- which aren't allowed in geographic distance
diff --git a/tests/gis_tests/utils.py b/tests/gis_tests/utils.py
index d2089e87a3..9e6002b01c 100644
--- a/tests/gis_tests/utils.py
+++ b/tests/gis_tests/utils.py
@@ -69,4 +69,5 @@ class FuncTestMixin:
 
     def tearDown(self):
         super().tearDown()
-        self.func_patcher.stop()
+        if self.func_patcher:
+            self.func_patcher.stop()
```

Reply:
Maybe it's better to keep the current version but catch the exception. The exception catching would be obsolete when dropping support for Python 3.7 (https://github.com/python/cpython/commit/02b84cb1b4f5407309c81c8b1ae0397355d6e568).